Fix Remove forward slash from desktop filename (Shenzhen I/O)

slowsage opened the pull request:
## Description

Installing Shenzhen I/O throws the following error:
```
Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/minigalaxy/installer.py", line 72, in install_game
    error_message = create_applications_file(game)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/minigalaxy/installer.py", line 271, in create_applications_file
    os.remove(path_to_shortcut)
FileNotFoundError: [Errno 2] No such file or directory: '/home/USERNAME/.local/share/applications/SHENZHEN I/O.desktop'

```
`/` is not a valid character for a filename in linux. This PR removes all `/` from the filename part of the path.
## Checklist
 
 - [x] _CHANGELOG.md_ was updated (**format**: - Change made (thanks to github_username))

sharkwouter reviewed (changes requested):
Thanks for this PR, that's a good find!

I did find one issue with this change. Could you add the same change to the `uninstall_game` function? There the desktop file is removed.

slowsage commented:
Can also add a one line helper function `_sanitize_filename` that is reused if thats cleaner. Not sure what you prefer.

sharkwouter commented:
If you want to use a sanitize method, I would suggest you do the following:
- Update this function to exclude slashes: https://github.com/sharkwouter/minigalaxy/blob/master/minigalaxy/game.py#L50
- Then use this function in your change: https://github.com/sharkwouter/minigalaxy/blob/master/minigalaxy/game.py#L22
- It probably makes sense then to also make this function use the _strip_string method: https://github.com/sharkwouter/minigalaxy/blob/master/minigalaxy/game.py#L25

slowsage commented:
`__strip_string` excludes slashes already but unfortunately also spaces which messes up the desktop file names. I have changed the code to reuse `get_stripped_name` in `installer.py` and updated `get_install_directory_name` since its basically the same.

sharkwouter commented:
Thanks for the fix! It's really appreciated!
